dataflow/flex-templates/streaming_beam/streaming_beam_test.py

```diff
@@ -0,0 +1,167 @@
+            b'{"url": "https://beam.apache.org/", "review": "positive"}')
+        future.result()
+        time.sleep(10)
+
+
+def test_dataflow_flex_templates_pubsub_to_bigquery():
+    # Create BigQuery client and dataset.
+    bigquery_client = bigquery.Client(project=PROJECT)
+    dataset_id = '{}.{}'.format(PROJECT, DATASET)
+    dataset = bigquery.Dataset(dataset_id)
+    dataset = bigquery_client.create_dataset(dataset, exists_ok=True)
+
+    # Create Pubsub topic and subscription.
+    topic_path = create_topic_path()
+    sub_path = create_subscription(topic_path)
+
+    # Use one process to publish messages to a topic.
+    publish_process = mp.Process(target=lambda: _infinite_publish_job(topic_path))
+
+    # Use another process to run the streaming pipeline that should write one
+    # row to BigQuery every minute (according to the default window size).
+    pipeline_process = mp.Process(target=lambda: sp.call([
+        'python',
+        'streaming_beam.py',
+        '--project',
+        PROJECT,
+        '--runner',
+        'DirectRunner',
+        '--temp_location',
+        tempfile.mkdtemp(),
+        '--input_subscription',
+        sub_path,
+        '--output_table',
+        '{}:{}.{}'.format(PROJECT, DATASET, TABLE),
+    ]))
+
+    publish_process.start()
+    pipeline_process.start()
+
+    pipeline_process.join(timeout=90)
+    publish_process.join(timeout=0)
+
+    pipeline_process.terminate()
+    publish_process.terminate()
+
+    # Check for output data in BigQuery.
+    time.sleep(10)
+    query = 'SELECT * FROM beam_samples.streaming_beam_sql'
+    query_job = bigquery_client.query(query)
+    rows = query_job.result()
+    assert rows.total_rows > 0
+    for row in rows:
+        assert row['score'] == 1
+
+    # Clean up.
+    bigquery_client.delete_table(
+        'beam_samples.streaming_beam_sql', not_found_ok=True)
+    bigquery_client.delete_dataset(DATASET, not_found_ok=True)
+    pubsub.PublisherClient().delete_topic(topic_path)
+    pubsub.SubscriberClient().delete_subscription(sub_path)
+
+
+# TODO:Testcase using Teststream currently does not work as intended.
+# The first write to BigQuery fails. Have filed a bug. The test case
+# to be changed once the bug gets fixed.
+'''
+@mock.patch("apache_beam.Pipeline", TestPipeline)
+@mock.patch(
+    "apache_beam.io.ReadFromPubSub",
+    lambda subscription: (
+        TestStream()
+        .advance_watermark_to(0)
+        .advance_processing_time(60)
+        .add_elements([TimestampedValue(
+            b'{"url": "https://beam.apache.org/", "review": "positive"}',
+                1575937195)])
+        .advance_processing_time(60)
+        .add_elements([TimestampedValue(
+            b'{"url": "https://beam.apache.org/", "review": "positive"}',
+                1575937255)])
+        .advance_watermark_to_infinity()
+    ),
+)
+def test_dataflow_flex_templates_pubsub_to_bigquery():
+
+    # Create BigQuery client, dataset and table.
```

Comment:
Thank you for keeping this code. This setup and the teardown code below should be taken care of by fixtures so it would be safe to remove. (But add the fixture "calls" at the test arguments)

Reply:
Done.